tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +798,460 @@ int main(int argc, char **argv)
             }
         }
 
+        /* Test ChaCha20 boosting behaviour for sslv2 connection */
+        {
+            security_policy.minimum_protocol_version = S2N_SSLv2;
+            DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(connection);
+
+            connection->security_policy_override = &security_policy;
+            connection->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+            EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(connection, S2N_SSLv2));
+            /* Setting this to tls12 just to bypass the higher_vers_match validation */
+            connection->actual_protocol_version = S2N_TLS12;
+
+            /* Need to set the server config for tls 1.2 ciphers so that we pass auth validation */
+            EXPECT_SUCCESS(s2n_connection_set_config(connection, server_config));
+
+            static struct s2n_cipher_suite *cipher_suites_sslv2[] = {
+                &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       /* 0xC0,0x23 */
+                &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256, /* 0xCC,0xA9 */
+                &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   /* 0xCC,0xA8 */
+                &s2n_dhe_rsa_with_chacha20_poly1305_sha256,     /* 0xCC,0xAA */
+            };
+
+            cipher_preferences = (struct s2n_cipher_preferences) {
+                .count = s2n_array_len(cipher_suites_sslv2),
+                .suites = cipher_suites_sslv2,
+                .allow_chacha20_boosting = true,
+            };
+
+            uint8_t ecdhe_ecdsa_chacha20_boosted_wire[] = {
+                0x00,
+                TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                0x00,
+                TLS_CHACHA20_POLY1305_SHA256,
+                0x00,
+                TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                0x00,
+                TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                0x00,
+                TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
+            };
+            uint8_t count = sizeof(ecdhe_ecdsa_chacha20_boosted_wire) / S2N_SSLv2_CIPHER_SUITE_LEN;
+
+            if (s2n_chacha20_poly1305.is_available()) {
+                /* Server and client have chacha20 boosting enabled. SSLv2 Server negotiates its most preferred
+                 * chacha20 cipher suite 
+                 */
+                {
+                    EXPECT_OK(s2n_toggle_chacha20_boosting(&cipher_preferences, true));
+                    cipher_preferences.allow_chacha20_boosting = true;
```

Comment:
what is the purpose of this line ? i thought s2n_toggle_chacha20_boosting already set cipher_preferences.allow_chacha20_boosting to true .

Reply:
It's redundant but I added this to be explicit. Also, this protects this test case since if another one was added before this and changed the flag then we would get different behaviour

Reply:
You must trust your function that toggles chacha boosting. Otherwise, here is no need for it ! Also, the name of the function is misleading. It is called toggle while in the function you are actually setting the flag to True or False based on the passed argument. So s2n_set_chacha20_boosting is more appropriate name .

Reply:
Oh I mis-read; I thought you were referring to line 848. I'll remove the redundant set. I can reword the function name.